modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/SabrParametersSwaptionVolatilities.java

```diff
@@ -0,0 +1,630 @@
+  @Override
+  public double volatility(double expiry, double tenor, double strike, double forwardRate) {
+    return parameters.volatility(expiry, tenor, strike, forwardRate);
+  }
+
+  @Override
+  public ValueDerivatives volatilityAdjoint(double expiry, double tenor, double strike, double forward) {
+    return parameters.volatilityAdjoint(expiry, tenor, strike, forward);
+  }
+
+  @Override
+  public double shift(double expiry, double tenor) {
+    return parameters.shift(expiry, tenor);
+  }
+
+  @Override
+  public SurfaceCurrencyParameterSensitivity surfaceCurrencyParameterSensitivity(SwaptionSensitivity pointSensitivity) {
+    throw new UnsupportedOperationException("Sensitivity is based on SwaptionSabrSensitivity, not SwaptionSensitivity");
+  }
+
+  /**
+   * Calculates the sensitivity to the nodes of the underlying volatility objects 
+   * <p>
+   * The underlying object is typically curve, surface or cube. 
+   * 
+   * @param sensitivity  the point sensitivity
+   * @return the node sensitivity
+   */
+  public SurfaceCurrencyParameterSensitivities surfaceCurrencyParameterSensitivity(SwaptionSabrSensitivity sensitivity) {
+    ArgChecker.isTrue(sensitivity.getConvention().equals(convention),
+        "Swap convention of provider must be the same as swap convention of swaption sensitivity");
+    double expiry = relativeTime(sensitivity.getExpiry());
+    double tenor = sensitivity.getTenor();
+    DoublesPair expiryTenor = DoublesPair.of(expiry, tenor);
+    SurfaceCurrencyParameterSensitivity alphaSensi = surfaceCurrencyParameterSensitivity(
+        parameters.getAlphaSurface(), sensitivity.getCurrency(), sensitivity.getAlphaSensitivity(), expiryTenor);
+    SurfaceCurrencyParameterSensitivity betaSensi = surfaceCurrencyParameterSensitivity(
+        parameters.getBetaSurface(), sensitivity.getCurrency(), sensitivity.getBetaSensitivity(), expiryTenor);
+    SurfaceCurrencyParameterSensitivity rhoSensi = surfaceCurrencyParameterSensitivity(
+        parameters.getRhoSurface(), sensitivity.getCurrency(), sensitivity.getRhoSensitivity(), expiryTenor);
+    SurfaceCurrencyParameterSensitivity nuSensi = surfaceCurrencyParameterSensitivity(
+        parameters.getNuSurface(), sensitivity.getCurrency(), sensitivity.getNuSensitivity(), expiryTenor);
+    return SurfaceCurrencyParameterSensitivities.of(alphaSensi, betaSensi, rhoSensi, nuSensi);
+  }
+
+  private SurfaceCurrencyParameterSensitivity surfaceCurrencyParameterSensitivity(
+      NodalSurface surface,
+      Currency currency,
+      double factor,
+      DoublesPair expiryTenor) {
+
+    Map<DoublesPair, Double> sensiMap = surface.zValueParameterSensitivity(expiryTenor);
+    return SurfaceCurrencyParameterSensitivity.of(
+        updateSurfaceMetadata(surface.getMetadata(), sensiMap.keySet()), currency,
+        DoubleArray.copyOf(sensiMap.values().parallelStream().map((p) -> (p) * factor).collect(Collectors.toList())));
```

Comment:
2 sets of unnecessary parens in `(p) -> (p) * factor`.

It would be more efficient to use

    sensiMap.values().parallelStream().mapToDouble(p -> p * factor).toArray()